fix comparator was using strings, not numbers to determine order

crccheck opened the pull request:
The old comparator was using the string values of the order attribute
instead of the numeric value, so you would get orders like:
0, 1, 10, 11, 2, 3, 4, 5, 6, 7, 8, 9

instead of
0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11

crccheck committed:
fix comparator was using ordering strings, not numbers

The old comparator was using the string values of the order attribute
instead of the numeric value, so you would get orders like:
0, 1, 10, 11, 2, 3, 4, 5, 6, 7, 8, 9

niran commented:
There's gotta be a more explicit way to do that... @tswicegood? parseInt?

tswicegood commented:
I'd prefer `parseInt`.  Well, I actually wish that this was ok:

```
String.prototype.toInt = function() {
  return parseInt(this);
};
```

But since modifying the global prototype generally isn't a good idea, let's just do `parseInt`.

crccheck commented:
the problem with parseInt is the line then becomes `return parseInt(node.get("order"), 10);` which I personally find to be less readable.

crccheck committed:
switch to a more explicit means of getting a number

in a jsperf test, parseInt was much faster. Didn't use parseFloat
because I might as well make it match the field type.

tswicegood commented:
This is merged in as of 12e147a27f366e461da2a9a370f0947b8f91a7ae.

Note for future bugfix PRs like this -- please create the branch off of the latest stable tag.  I rebased your changes on top of `v1.6.0` so it follows the correct history.  You can see the changes in the [network graph](https://github.com/armstrong/armstrong.core.arm_wells/network).
